### PR TITLE
Issue1244

### LINKED ISSUE #1244: Bug with `ClassOrInterfaceDeclaration.remove(MethodDeclaration)`
I'm parsing some source code, transforming it (removing some methods) and then rewriting it using `LexicalPreservingPrinter` to preserve the original formatting as close as possible. Works fine in general, but I stumbled upon some code for which this does not work. Here's a minimal test case:

    package de.topobyte;

    import java.io.IOException;
    import java.util.List;
    
    import com.github.javaparser.JavaParser;
    import com.github.javaparser.ast.CompilationUnit;
    import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
    import com.github.javaparser.ast.body.MethodDeclaration;
    import com.github.javaparser.printer.lexicalpreservation.LexicalPreservingPrinter;
    
    public class TestFailMinimal2
    {
    
        private static String text = "public class Foo {\n\n"
                + "// Some comment\n\n" // does work with only one \n
                + "public void writeExternal() {}\n" + "}";
    
        public static void main(String[] args) throws IOException
        {
            CompilationUnit originalCu = JavaParser.parse(text);
            CompilationUnit cu = LexicalPreservingPrinter.setup(originalCu);
    
            cu.findAll(ClassOrInterfaceDeclaration.class).stream().forEach(c -> {
                transform(c);
            });
        }
    
        private static void transform(ClassOrInterfaceDeclaration c)
        {
            List<MethodDeclaration> methods = c.getMethodsByName("writeExternal");
            for (MethodDeclaration method : methods) {
                c.remove(method);
            }
        }
    
    }

Here's what happens:

    Exception in thread "main" java.lang.UnsupportedOperationException: kept token(}) vs ChildTextElement{public void writeExternal() {
    }}
        at com.github.javaparser.printer.lexicalpreservation.Difference.apply(Difference.java:631)
        at com.github.javaparser.printer.lexicalpreservation.LexicalPreservingPrinter$2.concreteListChange(LexicalPreservingPrinter.java:192)
        at com.github.javaparser.ast.observer.PropagatingAstObserver.listChange(PropagatingAstObserver.java:72)
        at com.github.javaparser.ast.NodeList.lambda$notifyElementRemoved$2(NodeList.java:457)
        at java.util.ArrayList.forEach(ArrayList.java:1249)
        at com.github.javaparser.ast.NodeList.notifyElementRemoved(NodeList.java:457)
        at com.github.javaparser.ast.NodeList.remove(NodeList.java:142)
        at com.github.javaparser.ast.body.TypeDeclaration.remove(TypeDeclaration.java:164)
        at com.github.javaparser.ast.body.ClassOrInterfaceDeclaration.remove(ClassOrInterfaceDeclaration.java:202)
        at de.topobyte.TestFailMinimal2.transform(TestFailMinimal2.java:33)
        at de.topobyte.TestFailMinimal2.lambda$0(TestFailMinimal2.java:25)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
        at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
        at de.topobyte.TestFailMinimal2.main(TestFailMinimal2.java:24)

Here's a link to the line where the exception gets thrown:

https://github.com/javaparser/javaparser/blob/7c94cecfd6fee9608e37bd0262a155e222ef1c08/javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/Difference.java#L631

### PR DESCRIPTION
**Pull request by ftomassetti**
Fix #1244

### CODE REVIEW

File: javaparser-testing/src/test/java/com/github/javaparser/printer/lexicalpreservation/LexicalPreservingPrinterTest.java [hunk 1]
```diff
@@ -982,4 +982,24 @@ public void handleDeprecatedAnnotationAbstractClass() {
...
+        CompilationUnit cu = LexicalPreservingPrinter.setup(originalCu);
+
+        cu.findAll(ClassOrInterfaceDeclaration.class).stream().forEach(c -> {
+            List<MethodDeclaration> methods = c.getMethodsByName("writeExternal");
+            for (MethodDeclaration method : methods) {
+                c.remove(method);
+            }
+        });
+        assertEquals("public class Foo {" + EOL+
+                "// Some comment" + EOL+
```
Comment (matozoid) [inline javaparser-testing/src/test/java/com/github/javaparser/printer/lexicalpreservation/LexicalPreservingPrinterTest.java]: Check out `TestUtils.assertEqualsNoEol` :-)

**Comment by matozoid** [comment]
+1

**Comment by matozoid** [comment]
Oh, @ftomassetti , if you want to do me a favour: when you pull something, set the milestone to "next release" for the PR and any issues.

**Comment by ftomassetti** [comment]
Understood!

**Comment by matozoid** [comment]
Thanks!

**Pull request merged**